src/contracts.rs

```diff
@@ -109,44 +131,86 @@ impl TonContracts {
     /// Get address for contract deploying
     pub fn get_deploy_address(
         &self,
-        code: &[u8],
-        keys: &Ed25519KeyPair,
+        abi: &str,
+        image: &[u8],
+        init_params: Option<RunParameters>,
+        public_key: &Ed25519Public,
+        workchain_id: i32,
     ) -> TonResult<TonAddress> {
-        let result: TonAddress = Interop::json_request(
+        self.get_deploy_data(Some(abi), Some(image), init_params, public_key, Some(workchain_id))?
+            .address
+            .ok_or(TonErrorKind::InternalError("No address in result".to_owned()).into())
+    }
+
+    /// Get contract deploy data: image (state init), storage data and deploying address
+    pub fn get_deploy_data(
+        &self,
+        abi: Option<&str>,
+        image: Option<&[u8]>,
+        init_params: Option<RunParameters>,
+        public_key: &Ed25519Public,
+        workchain_id: Option<i32>,
+    ) -> TonResult<ResultOfGetDeployData> {
+        let abi = abi.map(|val| 
+                serde_json::from_str(val).map_err(|_| TonErrorKind::InvalidArg(val.to_owned())))
+            .transpose()?;
```

Comment:
Maybe format like this?
```
let abi = abi.map(|val| {
    serde_json::from_str(val).map_err(|_| TonErrorKind::InvalidArg(val.to_owned()))
}).transpose()?;
```
Or it is not a rust style)

Reply:
Done